### DISCUSSION TITLE
Reducing time for large point cloud interpolation

### DISCUSSION THREAD

**Discussion by kpan2** [pyvista/pyvista#4373] [Q&A]
Hello,

I have a stack of CT images that I have reoriented in space (each slice has its own plane vector). I have the xyz coordinate and label value of each 'pixel' as a PolyData dataset of 2,000,000 points. I am essentially doing the same thing as the 'Complex Interpolation' example on this page: https://docs.pyvista.org/version/stable/examples/01-filter/interpolate.html . Currently, it already takes up to 5-6 hours to complete the interpolation. These points are only a small portion of the actual medical data I want to process, so I expect computation time to increase exponentially.

My current settings for the interpolation function:
grid.interpolate(pdata, radius=0.1, sharpness=2, strategy='null_value',n_points=6)

Is there a way to reduce the computation time for interpolating even larger point clouds into a volume? I feel like it might only be possible by compromising the interpolation radius, which is not ideal for medical image reconstruction. Unfortunately, I cannot provide the data set I am working with for confidentiality reasons.

Thank you in advance!

**Comment by banesullivan** [comment]
So you're converting image data slices to a point cloud then interpolating to a volume? Can you provide some visualizations to demonstrate your process/workflow (input/output)?

**Reply by kpan2** [reply]
yep, that's correct. Actually the 'Complex Interpolation' example is quite accurate to what I'm doing, except instead of 'temperature' values at each point I have the label value for each pixel. Meaning, if my image has background, bone and muscle, each pixel will have a label of 0,1, or 2. Similarly, my data points are scattered in space like the 'probes' in the example.

My inputs:
cloud = a (N,3) array containing the xyz coordinates of every point in my image stack in space
pdata = pv.PolyData(cloud)
pdata['values'] = the label values of each point 

Then I create a grid = pv.UniformGrid() with some desired dimension and spacing, and do 
output = grid.interpolate(pdata, radius=0.1, sharpness =2 , strategy = 'null_value', n_points = 6)

**Reply by banesullivan** [reply]
I don't think this is the best approach... can you please share screenshots of your data? If they are slices to begin with, then maybe we can just generate image data directly from those slices. Or we can use better algorithms in VTK for handling the structure you're given rather than resorting to point cloud interpolation. I really need you to share screenshots of the data at various points in the workflow or share a minimally working example in code to help further
